karapace/instrumentation/prometheus.py

```diff
@@ -0,0 +1,100 @@
+"""
+karapace - prometheus instrumentation
+
+Copyright (c) 2024 Aiven Ltd
+See LICENSE for details
+"""
+# mypy: disable-error-code="call-overload"
```

Comment:
As the `aiohttp.web.Application` and `aiohttp.web.Request` objects both support [collections.abc.MutableMapping](https://docs.python.org/3/library/collections.abc.html#collections.abc.MutableMapping), we can set attributes at runtime, which is beneficial and used within the HTTP requests instrumentation. 
MyPy complains about missing `overloads` and as these object are within 3rd party library, we cannot update the declarations and thus I disable this error for the whole file.